src/ServiceControl.Infrastructure.RavenDB/Extensions.cs

```diff
@@ -2,12 +2,34 @@
 {
     using System;
     using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus.Logging;
     using Raven.Abstractions.Data;
+    using Raven.Client;
     using Raven.Database;
     using Raven.Json.Linq;
 
     public static class Extensions
     {
+        static readonly ILog Log = LogManager.GetLogger(typeof(Extensions));
+        public static async Task TestAllIndexesAndResetIfException(this IDocumentStore store)
+        {
+            foreach (var index in store.DatabaseCommands.GetStatistics().Indexes)
+            {
+                try
+                {
+                    using (var session = store.OpenAsyncSession())
+                    {
+                        await session.Advanced.AsyncDocumentQuery<object>(index.Name).Take(1).ToListAsync().ConfigureAwait(false);
+                    }
+                }
+                catch (Exception)
+                {
+                    Log.Info("Resetting errored index: " + index.Name);
```

Comment:
- I would say that it should be at least a Warning, potentially Error. 
 - I would also expand on the exception message, and say what we did: 'When trying to fetch 1 document from the index the exception was thrown: {log the exception}, We are assuming that an index could be corrupted and need to be reset.' ((In theory, the exception could be not related to a bad index, could be connectivity issues, disk space issue etc.))
 - Since we have access to the index, we can log some other information: 
Priority, IsInvalid, IndexingAttempts

Reply:
@WojcikMike  I applied the suggestions. If you prefer, we can still rename the function from TestAllIndexesAndResetIfException to TestAllIndexes and remove the `store.DatabaseCommands.ResetIndex(index.Name);` line. This way at least we are logging better information for both the user and the support engineer